doc/api/assert.md

```diff
@@ -8,410 +8,495 @@ used in application code via `require('assert')`. However, `assert` is not a
 
 ```js
 const assert = require('assert');
 
+// OK
 assert.equal(1, 1);
-  // OK, 1 == 1
+
+// OK
 assert.equal(1, '1');
-  // OK, 1 == '1'
 
+// Throws an exception: AssertionError: 1 == 2
 assert.equal(1, 2);
-  // AssertionError: 1 == 2
+
+// Throws an exception: AssertionError: { a: { b: 1 } } == { a: { b: 1 } }
 assert.equal({a: {b: 1}}, {a: {b: 1}});
-  //AssertionError: { a: { b: 1 } } == { a: { b: 1 } }
 ```
 
-If the values are not equal, an `AssertionError` is thrown with a `message`
-property set equal to the value of the `message` parameter. If the `message`
-parameter is undefined, a default error message is assigned.
-
 ## assert.fail(actual, expected, message, operator)
 <!-- YAML
 added: v0.1.21
 -->
 
-Throws an `AssertionError`. If `message` is falsy, the error message is set as
-the values of `actual` and `expected` separated by the provided `operator`.
-Otherwise, the error message is the value of `message`.
+* `actual` {mixed} A value to check
+* `expected` {mixed} The expected value of `actual`
+* `message` {String} The exception message to use. If not supplied, then a
+  generated message containing `actual` and `expected` is used.
+* `operator` {String} If `message` is not specified, this is a comparison operator
+  that is inserted between `actual` and `expected` in the generated exception
+  message.
+* Return: `undefined`
+
+Always throws an `AssertionError`.
+
+Examples:
 
 ```js
 const assert = require('assert');
 
+// Throws an exception: AssertionError: 1 > 2
 assert.fail(1, 2, undefined, '>');
-  // AssertionError: 1 > 2
 
+// Throws an exception: AssertionError: whoops
 assert.fail(1, 2, 'whoops', '>');
-  // AssertionError: whoops
 ```
 
 ## assert.ifError(value)
 <!-- YAML
 added: v0.1.97
 -->
 
-Throws `value` if `value` is truthy. This is useful when testing the `error`
-argument in callbacks.
+* `value` {mixed}
+* Return: `undefined`
+
+Throws `value` if `value` evaluates to `true`. This is useful when testing the
+error argument in callbacks.
+
+Examples:
 
 ```js
 const assert = require('assert');
 
-assert.ifError(0); // OK
+// OK
+assert.ifError(0);
+
+// Throws an exception: 1
 assert.ifError(1); // Throws 1
-assert.ifError('error'); // Throws 'error'
-assert.ifError(new Error()); // Throws Error
+
+// Throws an exception: error
+assert.ifError('error');
+
+// Throws an exception: Error
+assert.ifError(new Error());
 ```
 
 ## assert.notDeepEqual(actual, expected[, message])
 <!-- YAML
 added: v0.1.21
 -->
 
-Tests for any deep inequality. Opposite of [`assert.deepEqual()`][].
+* `actual` {mixed} A value to check
+* `expected` {mixed} The expected value of `actual`
+* `message` {String} The exception message to use when `actual` and `expected`
+  are equal. If not supplied, then a generated message containing `actual` and
+  `expected` is used.
+
+Tests *deep* inequality between `actual` and `expected` as determined by the
+'not equal' operator (`!=`). If `actual` and `expected` are equal, then
```

Comment:
Actually, a solution might be to just remove the descriptors entirely.

So this:

> 'not equal' operator (`!=`)

becomes this:

> `!=` operator